In [0]:
%python
from pyspark.sql.functions import *
#loading the dataset into a dataframe with the header of the csv included
c_data= spark.read.csv("/Volumes/teaching/datasets/assignment_2/task_1/ctg-studies.csv", header=True)

In [0]:
%python
#Exploring the dataframe
c_data.show(10)
display(c_data)

+-----------+--------------------+---------+------------------+--------------------+--------------------+--------------------+--------------------+----------+-----------+--------------+--------------------+----------+---------------+
| NCT Number|         Study Title|  Acronym|      Study Status|          Conditions|       Interventions|             Sponsor|       Collaborators|Enrollment|Funder Type|    Study Type|        Study Design|Start Date|Completion Date|
+-----------+--------------------+---------+------------------+--------------------+--------------------+--------------------+--------------------+----------+-----------+--------------+--------------------+----------+---------------+
|NCT01359735|Healing Effects o...|     NULL|         COMPLETED|Non-melanoma Skin...|BIOLOGICAL: HP802...|         Healthpoint|                NULL|         8|   INDUSTRY|INTERVENTIONAL|Allocation: RANDO...|   2011-05|        2012-05|
|NCT07071935|A Clinical Trial ...|EVENT ALS|NOT_YET_RECRUITING|A

NCT Number,Study Title,Acronym,Study Status,Conditions,Interventions,Sponsor,Collaborators,Enrollment,Funder Type,Study Type,Study Design,Start Date,Completion Date
NCT01359735,Healing Effects of HP802-247 Versus Antibiotic Ointment in Mohs Micrographic Surgery Patients,null,COMPLETED,Non-melanoma Skin Cancer,BIOLOGICAL: HP802-247|BIOLOGICAL: Bacitracin Ointment,Healthpoint,null,8,INDUSTRY,INTERVENTIONAL,Allocation: RANDOMIZED|Intervention Model: PARALLEL|Masking: NONE|Primary Purpose: TREATMENT,2011-05,2012-05
NCT07071935,A Clinical Trial of Early Ventilation in Amyotrophic Lateral Sclerosis (EVENT ALS),EVENT ALS,NOT_YET_RECRUITING,Amyotrophic Lateral Sclerosis (ALS)|Chronic Respiratory Failure|Neuromuscular Disease Patients|Neuromuscular Disease|Respiratory Insufficiency|Respiratory Insufficiency Requiring Mechanical Ventilation|Positive Pressure Ventilation|Non-invasive Ventilation|Non-invasive Ventilation Support|Non-invasive Ventilatory Support,DEVICE: Non-invasive ventilation,University of Pennsylvania,United States Department of Defense,48,OTHER,INTERVENTIONAL,Allocation: RANDOMIZED|Intervention Model: PARALLEL|Masking: SINGLE (OUTCOMES_ASSESSOR)|Primary Purpose: TREATMENT,2026-03,2029-05
NCT02534935,"Immunogenicity, Safety and Tolerability of a Neisseria Meningitidis Serogroup B Bivalent Recominant Lipoprotein 2086 Vaccine (Bivalent rLP2086) in Healthy Toddlers.",null,COMPLETED,Meningococcal B Disease,BIOLOGICAL: rLP2086 vaccine|BIOLOGICAL: Pediatric HAV vaccine,Pfizer,null,396,INDUSTRY,INTERVENTIONAL,"Allocation: RANDOMIZED|Intervention Model: PARALLEL|Masking: TRIPLE (PARTICIPANT, CARE_PROVIDER, INVESTIGATOR)|Primary Purpose: PREVENTION",2015-08-31,2020-03-17
NCT05617235,Kinesiology Taping After Thoracotomy,Taping,COMPLETED,Post Operative Pain|Thoracotomy|Pulmonary Function,DEVICE: Kinesiology taping,Akdeniz University,null,88,OTHER,INTERVENTIONAL,Allocation: RANDOMIZED|Intervention Model: PARALLEL|Masking: SINGLE (PARTICIPANT)|Primary Purpose: PREVENTION,2019-02-06,2022-11-01
NCT00760435,Infliximab Plus Intravenous Immunoglobulin for the Primary Treatment of Kawasaki Disease,null,COMPLETED,Kawasaki Disease,DRUG: Infliximab|DRUG: Placebo,"University of California, San Diego",Nationwide Children's Hospital,196,OTHER,INTERVENTIONAL,"Allocation: RANDOMIZED|Intervention Model: PARALLEL|Masking: QUADRUPLE (PARTICIPANT, CARE_PROVIDER, INVESTIGATOR, OUTCOMES_ASSESSOR)|Primary Purpose: TREATMENT",2009-03,2012-10
NCT03711435,Traditional Chinese Medicine(TCM) Syndrome Classification of Idiopathic Pulmonary Fibrosis(IPF) and Metabolomics,null,COMPLETED,Idiopathic Pulmonary Fibrosis,null,Xuzhou Traditional Chinese Medicine Hospital,null,75,OTHER,OBSERVATIONAL,Observational Model: |Time Perspective: p,2018-10-20,2020-02-05
NCT00367835,SPD503 (Guanfacine Hydrochloride) in ADHD Plus Oppositional Symptoms,null,COMPLETED,ADHD,DRUG: SPD503 (Guanfacine hydrochloride)|DRUG: Placebo,Shire,null,217,INDUSTRY,INTERVENTIONAL,"Allocation: RANDOMIZED|Intervention Model: PARALLEL|Masking: QUADRUPLE (PARTICIPANT, CARE_PROVIDER, INVESTIGATOR, OUTCOMES_ASSESSOR)|Primary Purpose: TREATMENT",2006-12-04,2008-01-04
NCT05882435,Use of Augmented Intelligence for the Interpretation of Bone Standard X-rays Prescribed by the Emergency Department (IMMEDIAT Urgences),IMMEDIAT,UNKNOWN,Bone Fracture,OTHER: Organization with AI,Assistance Publique - Hôpitaux de Paris,null,8400,OTHER,INTERVENTIONAL,Allocation: RANDOMIZED|Intervention Model: CROSSOVER|Masking: NONE|Primary Purpose: HEALTH_SERVICES_RESEARCH,2023-05-31,2023-12-15
NCT06366035,LoVE4MUM: Virtual Engagement for Preventing Postpartum Depression,LoVE4MUM,NOT_YET_RECRUITING,Postpartum Depression|Maternal Behavior|mHealth Intervention|Mental Health Issue,DEVICE: LoVE4MUM mobile application,"National Institutes of Health, Ministry of Health Malaysia",null,64,OTHER,INTERVENTIONAL,Allocation: RANDOMIZED|Intervention Model: PARALLEL|Masking: SINGLE (INVESTIGATOR)|Primary Purpose: PREVENTION,2024-04-22,2024-10-

In [0]:
%python
#creating a temporary table from our dataframe so that we can query and manipulate the data
c_data.createOrReplaceTempView("trials")

In [0]:
--Showing the first 10 rows of the table
SELECT * FROM trials LIMIT(10)

NCT Number,Study Title,Acronym,Study Status,Conditions,Interventions,Sponsor,Collaborators,Enrollment,Funder Type,Study Type,Study Design,Start Date,Completion Date
NCT01359735,Healing Effects of HP802-247 Versus Antibiotic Ointment in Mohs Micrographic Surgery Patients,null,COMPLETED,Non-melanoma Skin Cancer,BIOLOGICAL: HP802-247|BIOLOGICAL: Bacitracin Ointment,Healthpoint,null,8,INDUSTRY,INTERVENTIONAL,Allocation: RANDOMIZED|Intervention Model: PARALLEL|Masking: NONE|Primary Purpose: TREATMENT,2011-05,2012-05
NCT07071935,A Clinical Trial of Early Ventilation in Amyotrophic Lateral Sclerosis (EVENT ALS),EVENT ALS,NOT_YET_RECRUITING,Amyotrophic Lateral Sclerosis (ALS)|Chronic Respiratory Failure|Neuromuscular Disease Patients|Neuromuscular Disease|Respiratory Insufficiency|Respiratory Insufficiency Requiring Mechanical Ventilation|Positive Pressure Ventilation|Non-invasive Ventilation|Non-invasive Ventilation Support|Non-invasive Ventilatory Support,DEVICE: Non-invasive ventilation,University of Pennsylvania,United States Department of Defense,48,OTHER,INTERVENTIONAL,Allocation: RANDOMIZED|Intervention Model: PARALLEL|Masking: SINGLE (OUTCOMES_ASSESSOR)|Primary Purpose: TREATMENT,2026-03,2029-05
NCT02534935,"Immunogenicity, Safety and Tolerability of a Neisseria Meningitidis Serogroup B Bivalent Recominant Lipoprotein 2086 Vaccine (Bivalent rLP2086) in Healthy Toddlers.",null,COMPLETED,Meningococcal B Disease,BIOLOGICAL: rLP2086 vaccine|BIOLOGICAL: Pediatric HAV vaccine,Pfizer,null,396,INDUSTRY,INTERVENTIONAL,"Allocation: RANDOMIZED|Intervention Model: PARALLEL|Masking: TRIPLE (PARTICIPANT, CARE_PROVIDER, INVESTIGATOR)|Primary Purpose: PREVENTION",2015-08-31,2020-03-17
NCT05617235,Kinesiology Taping After Thoracotomy,Taping,COMPLETED,Post Operative Pain|Thoracotomy|Pulmonary Function,DEVICE: Kinesiology taping,Akdeniz University,null,88,OTHER,INTERVENTIONAL,Allocation: RANDOMIZED|Intervention Model: PARALLEL|Masking: SINGLE (PARTICIPANT)|Primary Purpose: PREVENTION,2019-02-06,2022-11-01
NCT00760435,Infliximab Plus Intravenous Immunoglobulin for the Primary Treatment of Kawasaki Disease,null,COMPLETED,Kawasaki Disease,DRUG: Infliximab|DRUG: Placebo,"University of California, San Diego",Nationwide Children's Hospital,196,OTHER,INTERVENTIONAL,"Allocation: RANDOMIZED|Intervention Model: PARALLEL|Masking: QUADRUPLE (PARTICIPANT, CARE_PROVIDER, INVESTIGATOR, OUTCOMES_ASSESSOR)|Primary Purpose: TREATMENT",2009-03,2012-10
NCT03711435,Traditional Chinese Medicine(TCM) Syndrome Classification of Idiopathic Pulmonary Fibrosis(IPF) and Metabolomics,null,COMPLETED,Idiopathic Pulmonary Fibrosis,null,Xuzhou Traditional Chinese Medicine Hospital,null,75,OTHER,OBSERVATIONAL,Observational Model: |Time Perspective: p,2018-10-20,2020-02-05
NCT00367835,SPD503 (Guanfacine Hydrochloride) in ADHD Plus Oppositional Symptoms,null,COMPLETED,ADHD,DRUG: SPD503 (Guanfacine hydrochloride)|DRUG: Placebo,Shire,null,217,INDUSTRY,INTERVENTIONAL,"Allocation: RANDOMIZED|Intervention Model: PARALLEL|Masking: QUADRUPLE (PARTICIPANT, CARE_PROVIDER, INVESTIGATOR, OUTCOMES_ASSESSOR)|Primary Purpose: TREATMENT",2006-12-04,2008-01-04
NCT05882435,Use of Augmented Intelligence for the Interpretation of Bone Standard X-rays Prescribed by the Emergency Department (IMMEDIAT Urgences),IMMEDIAT,UNKNOWN,Bone Fracture,OTHER: Organization with AI,Assistance Publique - Hôpitaux de Paris,null,8400,OTHER,INTERVENTIONAL,Allocation: RANDOMIZED|Intervention Model: CROSSOVER|Masking: NONE|Primary Purpose: HEALTH_SERVICES_RESEARCH,2023-05-31,2023-12-15
NCT06366035,LoVE4MUM: Virtual Engagement for Preventing Postpartum Depression,LoVE4MUM,NOT_YET_RECRUITING,Postpartum Depression|Maternal Behavior|mHealth Intervention|Mental Health Issue,DEVICE: LoVE4MUM mobile application,"National Institutes of Health, Ministry of Health Malaysia",null,64,OTHER,INTERVENTIONAL,Allocation: RANDOMIZED|Intervention Model: PARALLEL|Masking: SINGLE (INVESTIGATOR)|Primary Purpose: PREVENTION,2024-04-22,2024-10-

In [0]:
--Seperating the data in the study type column with "|" as the delimiter
CREATE OR REPLACE TEMP VIEW F_study AS SELECT split(`Study Type`,'\\|') as study_tp FROM trials

Question 1: Listing the common clinical trails as seen in the study type column and arranging them according to frequency of study. For this question, there is some data cleaning required and that is our first step. A temporary view is created with only the study type column. To better explore the data in this column, it has to be split and expanded so each row contains only a single condition. This is important since it will facilitate the counting function. For this task, Temporary views "```CREATE OR REPLACE TEMP VIEW```" were used to limit the access to only this notebook session. 

In [0]:
--Expanding the data that has been split in the new study_tp column and move them to new rows in the table
CREATE OR REPLACE TEMP VIEW E_study AS SELECT explode(study_tp) as Clinical_types FROM F_study

In [0]:
--Viewing the columns of the new table created with the expanded study type columns
SELECT * FROM E_study

Clinical_types
INTERVENTIONAL
INTERVENTIONAL
INTERVENTIONAL
INTERVENTIONAL
INTERVENTIONAL
OBSERVATIONAL
INTERVENTIONAL
INTERVENTIONAL
INTERVENTIONAL
INTERVENTIONAL


In [0]:
--Ranking the most frequent study types in our data
SELECT Clinical_types, COUNT(*) AS frequency
FROM E_study
GROUP BY Clinical_types
ORDER BY frequency DESC

Clinical_types,frequency
INTERVENTIONAL,437056
OBSERVATIONAL,133504
EXPANDED_ACCESS,1033
OTHER,162
INDUSTRY,23
OTHER_GOV,10
100,6
60,6
40,5
150,5


The studies are ranked in descending order and the most frequent study was discovered to be INTERVENTIONS with 437056 studies counted. The table shows some of the issues present in our dataset as some of the entries are not well defined studies. This is an issue with the original dataset that we are analysing. For this process the clinical_types are counted and grouped together  using ```SELECT clinical_types, COUNT(*) AS frequency``` and ```GROUP BY clinical_types```

Question 2: showing the top 10 conditions ranked by the number of times they appeared in the clinical trials. Same as the first approach for this question there will be some data cleaning. In the conditions column, Some columns have multiple entries for a single trial. The conditions column will be split using the delimiter "|" to create an array of the conditions. The array has to be exploded and each row will have a single condition. This is crucial since every entry will counted. The process of splitting and exploded the array is done in one step to make the code more effective. For this the code `explode(split(Conditions,'\\|'))` carries out this execution in a single step. One important detail is "Healthy" is among the conditions and this should be excluded when ranking the conditions. Before finalising the ranking, the code should know to exclude any entries with "Healthy" using `FROM Conditions_study WHERE ConditionsA != 'Healthy'`


In [0]:
--Splitting the data in the conditions column and exploding the column with delimiter "|"
CREATE OR REPLACE TEMP VIEW Conditions_study AS SELECT explode(split(Conditions,'\\|')) as ConditionsA FROM trials

In [0]:
--Exploring the new table with the expanded data in the conditions column
SELECT * FROM Conditions_study

ConditionsA
Non-melanoma Skin Cancer
Amyotrophic Lateral Sclerosis (ALS)
Chronic Respiratory Failure
Neuromuscular Disease Patients
Neuromuscular Disease
Respiratory Insufficiency
Respiratory Insufficiency Requiring Mechanical Ventilation
Positive Pressure Ventilation
Non-invasive Ventilation
Non-invasive Ventilation Support


In [0]:
SELECT ConditionsA, COUNT(*) AS frequency 
FROM Conditions_study WHERE ConditionsA != 'Healthy'
GROUP BY ConditionsA
ORDER BY frequency DESC LIMIT(10)

ConditionsA,frequency
Breast Cancer,8510
Obesity,7319
Stroke,5031
Hypertension,4508
Depression,4437
Pain,4392
Prostate Cancer,4306
HIV Infections,3847
Cancer,3836
Coronary Artery Disease,3642


From the rankings, The top 10 conditions being studied can be seen and number of studies are also added for deeper details. For this question the results were more cohesive and most of the conditions are well known and identifiable. This data could be used by companies when requesting funding for the listed conditions. The table shows the conditions that deserve priority and need more resources.


Question 3: The average months for the clinical trial for all completed studies. To calculate the average months for the clinical trial, The trials with a valid start date and completed date are selected. The date formats for both columns are start and completed columns need to be addressed. The dates in both columns are changed to a standard format 'yyyy-MM' which is year/month. After this step, The average months for the trial is calculated and displayed.The trails take an average of 35.02 months to complete according to the dataset.

In [0]:
--Aligning the data in the completion Date and  start date columns under the same format and calculating the average trial length in months
SELECT ROUND(AVG(months_between(
    to_date(LEFT(`Completion Date`, 7), 'yyyy-MM'),
    to_date(LEFT(`Start Date`, 7), 'yyyy-MM')
  )), 2) AS mean_trial_length_months
--The code only considers rows where both date columns are not null essentially only considering the trials that have been completed
FROM trials
WHERE `Completion Date` IS NOT NULL
  AND `Start Date` IS NOT NULL

mean_trial_length_months
35.02


Question 4:The number of alzheimer studies per year and the visualization of the trends. The temporary view is created with study status, completion date and conditions. The conditions are split up and exploded just like the previous processes. For this question, The specific condition under investigation is alzheimers and the aim is to show the trend of cases over the years. For the dates, The first 4 character of the completion date are selected and that's what is used to group all the present cases of alzheimers with the same year. This helps us visualize the trend of how the condition has developed over the years. To locate all cases of Alzheimer, the code `LOWER(ConditionsA) LIKE '%alzheimer%'` parses out all forms of alzheimer present in the dataset. This is a crucial step in filtering out all studies of Alzheimer.

In [0]:

--creating a table with study status, completion date and expanded conditions columns
CREATE OR REPLACE TEMP VIEW C_study AS SELECT `Study Status`, `Completion Date`, explode(split(Conditions,'\\|')) as ConditionsA FROM trials

In [0]:
SELECT * FROM C_study LIMIT(10)

Study Status,Completion Date,ConditionsA
COMPLETED,2012-05,Non-melanoma Skin Cancer
NOT_YET_RECRUITING,2029-05,Amyotrophic Lateral Sclerosis (ALS)
NOT_YET_RECRUITING,2029-05,Chronic Respiratory Failure
NOT_YET_RECRUITING,2029-05,Neuromuscular Disease Patients
NOT_YET_RECRUITING,2029-05,Neuromuscular Disease
NOT_YET_RECRUITING,2029-05,Respiratory Insufficiency
NOT_YET_RECRUITING,2029-05,Respiratory Insufficiency Requiring Mechanical Ventilation
NOT_YET_RECRUITING,2029-05,Positive Pressure Ventilation
NOT_YET_RECRUITING,2029-05,Non-invasive Ventilation
NOT_YET_RECRUITING,2029-05,Non-invasive Ventilation Support


In [0]:
--Selecting the completed studies with the Alzheimer's condition
CREATE OR REPLACE TEMP VIEW completed AS 
SELECT * FROM C_study 
WHERE `Completion Date` IS NOT NULL
  AND `Study Status` = 'COMPLETED'
  AND LOWER(ConditionsA) LIKE '%alzheimer%'

In [0]:
--Considering only the first 4 characters of the completion date which is the year, The number of studies are grouped and counted for each year
SELECT LEFT(`Completion Date`, 4) AS completion_year, COUNT(*) AS num_studies
FROM completed
GROUP BY LEFT(`Completion Date`, 4)
ORDER BY completion_year

completion_year,num_studies
1996,1
1997,3
1998,2
1999,2
2000,1
2001,9
2002,6
2003,11
2004,13
2005,33


Databricks visualization. Run in Databricks to view.

The visualization shows a growing trend which indicates any increase in Alzheimer cases over the years. The table also lists all the years and the number of cases for each year making it easy to interpret and make a report. 

conclusion
The dataset is explored using SQL and it enables us to find trends and crucial information to guide us in future clinical trials. The studies shows interventional studies are the most common studies being carried out in the trails.An interventional study tests an intervention — a potential drug, medical device, activity, or procedure in people. It is also observed that most trials take up to 35 months with breast cancer being the condition with the most cases present in the dataset. The final discovery was the growth of Alzheimer cases over the years with 2025 being the year with the most cases in the clinical trial. One of the issues faced was the study type column which had some entries that were not cleaned and actual study types. The data would benefit from collection of more accurate data.

